intellij/src/saros/intellij/eventhandler/filesystem/LocalFilesystemModificationHandler.java

```diff
@@ -367,35 +368,103 @@ private void generateResourceDeletionActivity(@NotNull VirtualFileEvent virtualF
       LOG.trace("Reacting before resource deletion: " + deletedVirtualFile);
     }
 
-    SPath path = VirtualFileConverter.convertToSPath(project, deletedVirtualFile);
+    if (deletedVirtualFile.isDirectory()) {
+      generateFolderDeletion(deletedVirtualFile);
+    } else {
+      generateFileDeletion(deletedVirtualFile);
+    }
+  }
+
+  /**
+   * Generates and dispatches a folder deletion activity for the deleted folder.
+   *
+   * @param deletedFolder the folder that was deleted
+   */
+  private void generateFolderDeletion(@NotNull VirtualFile deletedFolder) {
```

Comment:
The naming of the two new methods does not conform to the existing ones in this class.

Reply:
Well, it does match the existing naming convention for the split case (see the move methods), where the methods explicitly handling file/folder logic do not have the 'Activity' suffix.

I created it this way to differentiate between methods that were called by the handler from those that were called 'internally' in the handling logic. But you are right, a uniform naming for all methods is preferable.